app/controllers/divisions_controller.rb

```diff
@@ -8,29 +8,34 @@ def index_redirect
   end
 
   def index
+    @years = (Division.order(:date).first.date.year..Division.order(:date).last.date.year).to_a
+
     if params[:mpc] && params[:mpn]
-      electorate = params[:mpc].gsub("_", " ")
-      name = params[:mpn].gsub("_", " ")
+      @mpc = params[:mpc]
+      @mpn = params[:mpn]
+      @house = params[:house]
 
-      @member = Member.with_name(name)
-      @member = @member.in_house(params[:house])
-      @member = @member.where(constituency: electorate)
-      @member = @member.order(entered_house: :desc).first
+      @member = get_member
 
-      if @member.nil?
-        render 'members/member_not_found', status: 404
-      else
+      begin
+        @date_start, @date_end = get_date_range(params[:date])
+      rescue ArgumentError
+        render 'home/error_404', status: 404
+      end
+
+      if @member
+        @divisions = @member.divisions_possible(@date_start, @date_end).order(date: :desc, clock_time: :desc, name: :asc)
         render 'index_with_member'
+      else
+        render 'members/member_not_found', status: 404
       end
     else
-      @years = (Division.order(:date).first.date.year..Division.order(:date).last.date.year).to_a
       @sort = params[:sort]
       @rdisplay = params[:rdisplay]
       @house = params[:house] unless params[:house] == "all"
 
       begin
-        @date_start, @date_end, @date_range = DivisionParameterParser.get_date_range(params[:date]) if params[:date]
-        @date_start, @date_end, @date_range = DivisionParameterParser.get_date_range(@years.last.to_s) if @date_start.nil? && @date_end.nil? && @rdisplay.nil?
+        @date_start, @date_end, @date_range = get_date_range(params[:date], @rdisplay)
```

Comment:
> I am not sure what this @rdisplay does. 

Looking at https://github.com/openaustralia/publicwhip/blob/d27b8db198087f4b7b6b5f33905f6c8550c5b382/app/models/parliament.rb it appears to be a way of selecting different parliaments (a parliament is kinda a period of time, between to elections. So after each election they say, "looks like the 57th Parliament is going to be a cracker mate!" It's definitely insider jargon and should be explained in a comment on the model). Like I want to see divisions from the 2007 Parliament (2008-2010) https://theyvoteforyou.org.au/divisions?rdisplay=2007 what a confusing name "rdisplay"?

We don't expose that in the UI anywhere that I know of. We've gone for the more intuitive years in the pagination which I think suits people's mental model better. I can see how this could be useful though.

I'm not sure about removing it. It could be worth opening an issue for it. Alternatives could be to add it to the API docs or the UI somehow in a discrete way. Or rename it something useful so devs know what this is and add a comment to the Parliament model :) That feels like a separate PR.

Does that help?